beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,229 @@
+    // hence we need to multiply the result by the scrape interval
+    // (set in the config file) and divide by the uptime search range.
+    const paramsUptime = {
+      query: `sum_over_time(up{instance='${peer.metric.instance}', job='${prometheous_job}'}[${uptimeSearchRange}s] offset ${offset}s) * ${scrapeInterval} / ${uptimeSearchRange}`,
+    };
+
+    const resultUptime = await queryPrometheus(queryUptime, paramsUptime);
+    const resultUptimePercent = resultUptime.data.result[0].value[1] * 100;
+    const upFactor = resultUptimePercent < requiredUptime ? 0 : 1;
+    peerData.set(factors.upTime, upFactor);
+    const upFactorCoefficient = upFactor ? uptimeSearchRange/rewardsInterval : 0
+    // Rewards should be adjusted by the upFactorCoefficient for a given peer if
+    // that peer joins the network later relative to the rewards interval start.
+    // Ex. if a peer joins mid month and all other factors are satisfied, then
+    // the rewards are devided by half.
+    peerData.set(upTimeRewardsCoefficient, upFactorCoefficient)
+
+    /// Pre-param requirement
+
+    const queryPreParams = `${prometheusAPI}/query`;
+    // <func>(<metric>{<labels>}[<local_range>] offset <time>)[<global_range>:<resolution>]
+    const paramsPreParams = {
+      query: `avg_over_time(tbtc_pre_params_count{instance='${peer.metric.instance}', job='${prometheous_job}'}[${preParamsAvgInterval}h] offset ${offset}s)[${rewardsInterval}s:${preParamsResolution}h]`,
+    };
+
+    const resultPreParams = await queryPrometheus(queryPreParams, paramsPreParams)
+    peerData.set(factors.preParams, 1);
+    if (resultPreParams.data.result.length == 0) {
+      peerData.set(factors.preParams, 0);
+    } else {
+      resultPreParams.data.result[0].values.forEach(function (peerPreParams: any) {
+        if (Number(peerPreParams[1]) < minPreParams) {
+          peerData.set(factors.preParams, 0);
+        }
+      })
+    }
+
+    /// Version requirement (One-week delay in updates to the most recent version)
+
+    // TODO: implement
+    // latest tag: git describe --tags $(git rev-list --tags --max-count=1)
+    // unix timestamp: git show -s --format=%ct v1.3.1-6464-g43ea52158
+
+
+    peersData.set(peer.metric.instance, peerData)
+    console.log("peersDataFactors", peersData)
+    console.log("counter: ", counter++)
+  }
+
+  // TODO: calculate rewards for a given address
+  //
+  // for a given peer:
+  // - check if all the requirements were satisfied (factors.* == 1)
+  // - if all the reqs ^ are satisfied calculate the rewards:
+  // -- peerRewards = (peer's authorized stake / total authorized stake) * rewardsForAGivenMonth * upTimeRewardsCoefficient
+
+  const jsonObject = await convertToJSON(peersData);
+  // Save to file
+  fs.writeFileSync(
+    peersDataFile,
+    JSON.stringify(jsonObject, null, 2)
+  )
+}
+
+async function convertToJSON(map: Map<string, Map<string, any>>) {
+  var json: {[k:string]: any} = {};
```

Comment:
```suggestion
  let json: {[k:string]: any} = {};
```
doesn't matter here, but i think it's good practice in general to be in the habit of using `let` instead of `var`. `var` has a really strange scope: https://stackoverflow.com/questions/762011/what-is-the-difference-between-let-and-var

Reply:
lol.. I think I was experimenting with smth and just straight-up forgot to change it. Yes, I changed it back to `let` as part of [982a1f0](https://github.com/keep-network/keep-core/pull/3346/commits/982a1f01cbd3036fa4c035d697edb323b75493c6)